### PR TITLE
Django Tutorial Part 8: User authentication and permissions

### PR DESCRIPTION
**Pull request by tdanhjv**
ready

### CODE REVIEW

**Comment by tdanhjv** [comment]
ready

File: locallibrary/catalog/models.py [hunk 1]
```diff
@@ -46,12 +48,20 @@ class BookInstance(models.Model):
...
+    borrower = models.ForeignKey(User, on_delete=models.SET_NULL, null=True, blank=True)
     LOAN_STATUS = (('m', _('Maintenance')), ('o', _('On loan')), ('a', _('Available')), ('r', _('Reserved')),)
     status = models.CharField(max_length=1, choices=LOAN_STATUS, blank=True, default='m',
                               help_text=_('Book availability', ))
 
+    @property
+    def is_overdue(self):
+        if self.due_back and date.today() > self.due_back:
+            return True
+        return False
```
Comment (huongnt-2545) [inline locallibrary/catalog/models.py]:
```suggestion
        return self.due_back and date.today() > self.due_back
```

File: locallibrary/catalog/templates/catalog/bookinstance_list_borrowed_user.html [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+        {% for bookinst in bookinstance_list %}
+        <li class="{% if bookinst.is_overdue %}text-danger{% endif %}">
+            <a href="{% url 'book-detail' bookinst.book.pk %}">{{ bookinst.book.title }}</a> ({{ bookinst.due_back }})
+        </li>
+        {% endfor %}
+    </ul>
+    {% else %}
+        <p>{% trans "There are no books borrowed." %}</p>
+    {% endif %}
+{% endblock %}
```
Comment (huongnt-2545) [inline locallibrary/catalog/templates/catalog/bookinstance_list_borrowed_user.html:19]: end file -> thêm dòng trống cuối file

File: locallibrary/templates/registration/logged_out.html [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{% extends "base_generic.html" %}
+{% load i18n %}
+
+{% block content %}
+    <p>{% trans "Logged out!" %}</p>
+    <a href="{% url 'login' %}">{% trans "Click here to login again." %}</a>
+{% endblock %}
```
Comment (huongnt-2545) [inline locallibrary/templates/registration/logged_out.html]: end line
sửa các file bên dưới nữa e nhé

File: locallibrary/catalog/models.py (hunk 1 above)
Comment (tdanhjv) [reply locallibrary/catalog/models.py]: resolved

File: locallibrary/catalog/templates/catalog/bookinstance_list_borrowed_user.html (hunk 2 above)
Comment (tdanhjv) [reply locallibrary/catalog/templates/catalog/bookinstance_list_borrowed_user.html:19]: resolved

File: locallibrary/templates/registration/logged_out.html (hunk 3 above)
Comment (tdanhjv) [reply locallibrary/templates/registration/logged_out.html]: resolved

**Comment by tdanhjv** [comment]
ready

**Review by huongnt-2545 — APPROVED** [review]
mớt pull nhé

**Comment by huongnt-2545** [comment]
merged

**Pull request merged**